### PR TITLE
Add branches scene

### PR DESCRIPTION
**Pull request by ELD**
Allow users to click on a branch to view the roster listing

#### Before you submit this PR for review, make sure the following tasks are complete.

- [x] Added relevant documentation.
- [x] Updated unit tests.
- [x] Made sure all tests are passing with `yarn test`
- [x] Ensured all code meets the [standard JS](https://standardjs.com) style guide.
- [x] Removed debug code.
- [x] Included screenshot if applicable.
- [x] Squashed changes into a single commit.
- [x] If adding new routes to the frontend application, specify them below for testing

    ```
    git rebase -i $(git merge-base HEAD master)
    git push origin NAME_OF_BRANCH --force-with-lease
    ```

image

### CODE REVIEW

File: titan-server/migrations/2018-12-09-200400_create_user_files_entry_types_table/up.sql [hunk 1]
```diff
@@ -1,6 +1,6 @@
 CREATE TABLE IF NOT EXISTS user_file_entry_types (
   id INT PRIMARY KEY AUTO_INCREMENT,
-  name VARCHAR(255) UNIQUE
+  name VARCHAR(255) UNIQUE NOT NULL
 );
 
 INSERT INTO user_file_entry_types(id, name) VALUES (23, ' Complete Course');
```
Comment (MichaelJamesParsons) [inline titan-server/migrations/2018-12-09-200400_create_user_files_entry_types_table/up.sql]: I recognize you didn't make this change, but the extra space in front of `Complete Course` should be removed.

File: titan-web-client/src/http/OrganizationsService.js [hunk 2]
```diff
@@ -6,6 +6,15 @@ class OrganizationsService {
     this.httpClient = this.httpService.getHttpClient();
   }
 
+  /**
+   * Get all organizations that the server knows about
```
Comment (MichaelJamesParsons) [inline titan-web-client/src/http/OrganizationsService.js]: nit: add `.` at the end of this description.

File: titan-web-client/src/layouts/dashboard/components/sidebar/SidebarHeading.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ class SidebarHeading extends React.Component {
 }
 
 SidebarHeading.propTypes = {
-  children: PropTypes.object
+  children: PropTypes.string
```
Comment (MichaelJamesParsons) [inline titan-web-client/src/layouts/dashboard/components/sidebar/SidebarHeading.js]: `children`  shouldn't be restricted to a string value. We may want to render more complex components as the heading text (i.e notification indicator, expand/collapse toggle, etc.).

File: titan-web-client/src/layouts/dashboard/components/sidebar/SidebarContentGroup.js [hunk 4]
```diff
@@ -17,7 +17,7 @@ class SidebarContentGroup extends React.Component {
 }
 
 SidebarContentGroup.propTypes = {
-  children: PropTypes.object
+  children: PropTypes.any
```
Comment (MichaelJamesParsons) [inline titan-web-client/src/layouts/dashboard/components/sidebar/SidebarContentGroup.js]: I don't believe `children` needs to be included in propTypes anymore. There should already be a lint rule to allow its usage without specifying it.

File: titan-web-client/src/modules/roster/branches/index.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import BranchCardGrid from '../components/BranchCardGrid';
+import OrganizationsService from '../../../http/OrganizationsService';
+
+class BranchesScene extends React.Component {
+  constructor (props) {
+    super(props);
+
+    this.organizationHttpService = new OrganizationsService();
+    this.state = {
+      branches: [
```
Comment (MichaelJamesParsons) [inline titan-web-client/src/modules/roster/branches/index.js]: Is this left over debug/playground code? These fields should be provided from the server. Setting `branches` as an empty array should be sufficient. We can add a loading spinner at a later time if necessary.

File: titan-web-client/src/modules/roster/branches/index.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+  componentDidMount () {
+    this.organizationHttpService.getAll()
+      .then((res) => {
+        this.setState({
+          branches: res.data
+        });
+      })
+      .catch((err) => {
+        console.log(err);
```
Comment (MichaelJamesParsons) [inline titan-web-client/src/modules/roster/branches/index.js]: Add a `FIXME` doc above this console log so we know to add better error handling later.

```
// FIXME update UI to indicate error state.
console.log(err);
```

File: titan-web-client/src/modules/roster/components/BranchCardGrid.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+import PropTypes from 'prop-types';
+import Column from 'titan/components/Grid/Column';
+import Row from 'titan/components/Grid/Row';
+import BranchCard from 'titan/modules/roster/components/BranchCard';
+
+/**
+ * Lists roster member cards in a grid.
+ */
+class BranchCardGrid extends React.Component {
+  constructor (props) {
```
Comment (MichaelJamesParsons) [inline titan-web-client/src/modules/roster/components/BranchCardGrid.js]: This constructor can be removed.

File: titan-web-client/src/modules/roster/components/BranchCardGrid.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    super(props);
+  }
+
+  render () {
+    return (
+      <Row wrap="wrap">
+        {this.props.branches.filter((e) => e.is_enabled === true).map((branch, index) => (
```
Comment (MichaelJamesParsons) [inline titan-web-client/src/modules/roster/components/BranchCardGrid.js]: This expression can be simplified:

- `is_enabled` holds a boolean value.
- in idiomatic JS,  single parameter functions don't need parentheses.

```
this.props.branches.filter(e => e.is_enabled)
```

File: titan-web-client/src/layouts/dashboard/components/sidebar/SidebarHeading.js (hunk 3 above)
Comment (MichaelJamesParsons) [reply titan-web-client/src/layouts/dashboard/components/sidebar/SidebarHeading.js]: As the following review mentions, `children` shouldn't need to be provided here anymore.

**Comment by ELD** [comment]
@MichaelJamesParsons Went ahead and resolved your comments. The component state was something left over from development. Started with component-local state before pulling it down from the API server.

File: titan-web-client/src/http/OrganizationsService.js [hunk 9]
```diff
@@ -6,6 +6,15 @@ class OrganizationsService {
     this.httpClient = this.httpService.getHttpClient();
   }
 
+  /**
+   * Get all organizations that the server knows about.
+   *
+   * @returns {Promise}
+   */
+  getAll () {
```
Comment (MichaelJamesParsons) [inline titan-web-client/src/http/OrganizationsService.js]: Rename to `findAll` to be consistent with other service clients.

File: titan-server/src/organizations/organizations.rs [hunk 10]
```diff
@@ -29,3 +29,7 @@ pub fn find_users(
 
     users_query.load::<(models::WcfUser, models::WcfUserAvatar)>(&**unkso_main)
 }
+
```
Comment (MichaelJamesParsons) [inline titan-server/src/organizations/organizations.rs]: Rename to `find_all` to be consistent with other DB methods.

**Comment by ELD** [comment]
@MichaelJamesParsons Resolved the last comments.

I'm going to cut the release for tonight from this branch as it's up to date with master. I'll wait to merge this in, though, until your final approval.

**Pull request merged**